storage/storage.go

```diff
@@ -137,6 +137,11 @@ func (s *Storage) Read(txn Transaction, path ast.Ref) (interface{}, error) {
 	return doc, nil
 }
 
+// Write updates a value in storage.
+func (s *Storage) Write(txn Transaction, op PatchOp, ref ast.Ref, value interface{}) error {
+	return s.builtin.Write(txn, op, ref, value)
+}
+
```

Comment:
Right, for now we don't support normal sense (as you described)

I've added a txn param to the Write  call so that (1) locking is handled in one place (i.e., when the transaction is created) and (2) if/when we eventually support transactions in the normal sense (as you described), we changes to storage users will be reduced (b/c they're already creating transactions to do any I/O against storage.)

If/when we introduce normal transactions, we'll need to add have a Commit and Rollback interface added like you described.